File: vercel-deploy/test_deployed_api.py
#!/usr/bin/env python3
"""
배포된 Vercel API 테스트 스크립트
실제 배포된 URL에서 API 기능을 테스트합니다.
"""

import requests
import json
import time

# 배포된 URL
BASE_URL = "https://vercel-deploy-671u6fysq-dal-kims-projects.vercel.app"

def test_submit_application():
    """신청서 제출 API 테스트"""
    print("🧪 신청서 제출 API 테스트 시작...")
    
    test_data = {
        "company_name": "테스트제조업체",
        "company_name_en": "Test Manufacturing Co.",
        "contact_name": "홍길동",
        "contact_email": "hong@test.com",
        "contact_phone": "010-1234-5678",
        "address": "서울시 강남구 테헤란로 123",
        "standards": ["ISO 9001", "ISO 14001"],
        "total_employees": 50,
        "site_count": 1,
        "part_time_count": 5,
        "contractor_count": 10,
        "shift_workers": 8,
        "is_integrated": True,
        "shared_management_system": True,
        "common_processes": True,
        "stage1": True,
        "stage2": True,
        "surveillance": True,
        "remote_audit_ratio": 0.2
    }
    
    try:
        response = requests.post(
            f"{BASE_URL}/api/submit-application",
            headers={"Content-Type": "application/json"},
            json=test_data,
            timeout=30
        )
        
        print(f"상태 코드: {response.status_code}")
        print(f"응답: {json.dumps(response.json(), ensure_ascii=False, indent=2)}")
        
        if response.status_code == 200:
            print("✅ 신청서 제출 API 테스트 성공!")
            return response.json()
        else:
            print("❌ 신청서 제출 API 테스트 실패!")
            return None
            
    except Exception as e:
        print(f"❌ 신청서 제출 API 테스트 오류: {str(e)}")
        return None

def test_create_quotation():
    """견적서 생성 API 테스트"""
    print("\n🧪 견적서 생성 API 테스트 시작...")
    
    test_data = {
        "company_name": "테스트제조업체",
        "company_name_en": "Test Manufacturing Co.",
        "contact_name": "홍길동",
        "contact_email": "hong@test.com",
        "contact_phone": "010-1234-5678",
        "address": "서울시 강남구 테헤란로 123",
        "standards": ["ISO 9001", "ISO 14001"],
        "total_employees": 50,
        "sites": [{
            "name": "본사",
            "address": "서울시 강남구 테헤란로 123",
            "total_headcount": 50,
            "part_time_count": 5,
            "contractor_count": 10,
            "shift_workers": 8,
            "standards": ["ISO 9001", "ISO 14001"]
        }],
        "integration": {
            "is_integrated": True,
            "shared_management_system": True,
            "common_processes": True,
            "same_audit_team": False
        },
        "options": {
            "stage1": True,
            "stage2": True,
            "surveillance": True,
            "recert": False,
            "remote_audit_ratio": 0.2,
            "day_rate": 1300000,
            "vat_rate": 0.1
        }
    }
    
    try:
        response = requests.post(
            f"{BASE_URL}/api/create-quotation",
            headers={"Content-Type": "application/json"},
            json=test_data,
            timeout=30
        )
        
        print(f"상태 코드: {response.status_code}")
        print(f"응답: {json.dumps(response.json(), ensure_ascii=False, indent=2)}")
        
        if response.status_code == 200:
            print("✅ 견적서 생성 API 테스트 성공!")
            return response.json()
        else:
            print("❌ 견적서 생성 API 테스트 실패!")
            return None
            
    except Exception as e:
        print(f"❌ 견적서 생성 API 테스트 오류: {str(e)}")
        return None

def test_send_email(quotation_data):
    """이메일 전송 API 테스트"""
    print("\n🧪 이메일 전송 API 테스트 시작...")
    
    test_data = {
        "recipient_email": "hong@test.com",
        "quotation": quotation_data.get('quotation', {})
    }
    
    try:
        response = requests.post(
            f"{BASE_URL}/api/send-email",
            headers={"Content-Type": "application/json"},
            json=test_data,
            timeout=30
        )
        
        print(f"상태 코드: {response.status_code}")
        print(f"응답: {json.dumps(response.json(), ensure_ascii=False, indent=2)}")
        
        if response.status_code == 200:
            print("✅ 이메일 전송 API 테스트 성공!")
            return response.json()
        else:
            print("❌ 이메일 전송 API 테스트 실패!")
            return None
            
    except Exception as e:
        print(f"❌ 이메일 전송 API 테스트 오류: {str(e)}")
        return None

def test_main_page():
    """메인 페이지 접근 테스트"""
    print("🧪 메인 페이지 접근 테스트 시작...")
    
    try:
        response = requests.get(f"{BASE_URL}/", timeout=30)
        
        print(f"상태 코드: {response.status_code}")
        
        if response.status_code == 200:
            print("✅ 메인 페이지 접근 성공!")
            print(f"페이지 크기: {len(response.text)} bytes")
            return True
        else:
            print("❌ 메인 페이지 접근 실패!")
            return False
            
    except Exception as e:
        print(f"❌ 메인 페이지 접근 오류: {str(e)}")
        return False

def main():
    """메인 테스트 함수"""
    print("🚀 배포된 Vercel API 테스트 시작")
    print("=" * 50)
    print(f"테스트 URL: {BASE_URL}")
    print("=" * 50)
    
    # 0. 메인 페이지 테스트
    main_page_success = test_main_page()
    
    # 1. 신청서 제출 테스트
    application_result = test_submit_application()
    
    # 2. 견적서 생성 테스트
    quotation_result = test_create_quotation()
    
    # 3. 이메일 전송 테스트 (견적서 데이터가 있는 경우)
    if quotation_result and quotation_result.get('success'):
        test_send_email(quotation_result)
    
    print("\n" + "=" * 50)
    print("🏁 API 테스트 완료")
    
    # 결과 요약
    if main_page_success and application_result and quotation_result:
        print("✅ 모든 테스트가 성공적으로 완료되었습니다!")
        print(f"🌐 웹사이트: {BASE_URL}")
        print(f"📊 견적 금액: ₩{quotation_result['quotation']['total_cost']:,}")
        print(f"📅 총 심사일수: {quotation_result['quotation']['total_audit_days']}일")
        print("\n🎉 Vercel 배포가 완전히 성공했습니다!")
    else:
        print("❌ 일부 테스트가 실패했습니다.")
        if not main_page_success:
            print("  - 메인 페이지 접근 실패")
        if not application_result:
            print("  - 신청서 제출 API 실패")
        if not quotation_result:
            print("  - 견적서 생성 API 실패")

if __name__ == "__main__":
    main()

File: vercel-deploy/test_deployed_api.js
/**
 * 배포된 Vercel API 테스트
 */

async function testDeployedAPI() {
  console.log('🚀 배포된 Vercel API 테스트 시작');
  console.log('='.repeat(50));
  
  // A회사 사례 (단일사업장)
  const testData = {
    client_name: 'A회사 (테스트)',
    sites: [{
      name: '울산공장',
      address: '울산광역시',
      standards: ['ISO14001'],
      regularEmployees: 40,
      partTimeEmployees: 10,
      business_sector: '화학',
      business_description: '나프탈렌 생산',
      management_system_maturity: 'MEDIUM'
    }],
    standards: ['ISO14001'],
    options: {
      stage1: true,
      stage2: true,
      surveillance: true,
      recert: true,
      integrated_audit: false,
      remote_audit: false
    }
  };

  try {
    console.log('📡 Vercel API 호출 중...');
    const response = await fetch('https://lrqa-iso-application.vercel.app/api/core-brain', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(testData)
    });

    if (!response.ok) {
      throw new Error(`HTTP ${response.status}: ${response.statusText}`);
    }

    const result = await response.json();
    
    console.log('✅ API 응답 성공!');
    console.log('\n📊 결과 요약:');
    console.log(`- 클라이언트: ${result.data.client_name}`);
    console.log(`- 총 심사일수: ${result.data.total_audit_days}일`);
    console.log(`- 총 비용: ${result.data.total_cost.toLocaleString()}원`);
    console.log(`- 일당: ${result.data.day_rate.toLocaleString()}원`);
    
    console.log('\n🏢 사업장별 상세:');
    result.data.breakdowns.forEach((breakdown, index) => {
      console.log(`  ${index + 1}. ${breakdown.site_name}:`);
      console.log(`     - ENP: ${breakdown.enp}명, 복잡도: ${breakdown.complexity}`);
      console.log(`     - EA 코드: ${breakdown.ea_code} (${breakdown.ea_matched_keyword})`);
      console.log(`     - 심사일수: ${breakdown.total_days}일`);
      console.log(`     - 비용: ${breakdown.cost.toLocaleString()}원`);
    });

    console.log('\n📋 가정사항:');
    result.data.assumptions.forEach(assumption => {
      console.log(`  - ${assumption.item}: ${assumption.description}`);
    });

    console.log('\n🎯 배포된 API가 정상 작동합니다!');
    
  } catch (error) {
    console.error('❌ API 호출 실패:', error.message);
    console.log('\n🔍 문제 해결 방법:');
    console.log('1. Vercel 배포가 완전히 완료되었는지 확인');
    console.log('2. API 엔드포인트 URL 확인');
    console.log('3. 네트워크 연결 상태 확인');
  }
}

// 테스트 실행
testDeployedAPI();